ElevenLabs-DotNet/Authentication/ElevenLabsClientSettings.cs

```diff
@@ -60,18 +59,39 @@ public ElevenLabsClientSettings(string domain, string apiVersion = DefaultApiVer
 
             Domain = $"{protocol}{domain}";
             ApiVersion = apiVersion;
-            BaseRequest = $"/{ApiVersion}/";
-            BaseRequestUrlFormat = $"{Domain}{BaseRequest}{{0}}";
+            BaseVersionedRequestUrlFormat = $"{Domain}/{{1}}/{{0}}";
         }
 
         public string Domain { get; }
 
         public string ApiVersion { get; }
 
-        public string BaseRequest { get; }
-
-        public string BaseRequestUrlFormat { get; }
+        /// <summary>
+        /// String with interpolation for the endpoint name (0) and api version (1).
+        /// </summary>
+        private string BaseVersionedRequestUrlFormat { get; }
 
         public static ElevenLabsClientSettings Default { get; } = new();
+
+        /// <summary>
+        /// Build a according to this settings with the given endpoint and api version.
+        /// </summary>
+        /// <param name="endpoint">The endpoint to build the url for (required).</param>
+        /// <param name="apiVersion">The version of the ElevenLabs api you want to use (optional).</param>
+        /// <returns>A string representing the built URL.</returns>
+        /// <exception cref="ArgumentException">Thrown when the endpoint is null or empty.</exception>
+        public string BuildUrl(string endpoint, string apiVersion = null)
```

Comment:
Not sure I like this pattern as much. Would prefer to keep it in the base endpoint class.

Was this to resolve the proxy support with versioned endpoints?

Reply:
No, it was just for just personal preference so no problem